### DISCUSSION TITLE
In uvicorn --reload mode, when I make save a new change in file it runs but first shows lot of errors

### DISCUSSION THREAD

**Discussion by angad-singhh** [fastapi/fastapi#11535] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root():
    return {"hi form first page"}

@app.post("/home")
def home():
    return{"hi from home page"}
```

### Description

when i run the main.py file through command ( uvicorn main:app ), it runs perfectly, but when i try to use ( uvicorn main:app --reload ), at first run it runs perfectly but when i change a file and save it ---> it shows errors in the terminal first then it runs, it happens every time i save a change, error shows then it runs, this makes the terminal very clotty.

this is the error that occur every time i save a change:

![image](https://github.com/tiangolo/fastapi/assets/108326021/55a8c1de-8863-414d-9624-5e1c363bf3e7)

### Operating System

Windows

### Operating System Details

![image](https://github.com/tiangolo/fastapi/assets/108326021/ef8acb11-1ccc-41a1-a067-5fc05ce81b93)

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

3.12.3

**Comment by YuriiMotov** [comment, marked as answer]
I tried it on Ubuntu 20.04.6, Python 3.12.2, FastAPI 0.111.0, Uvicorn 0.29.0. And I don't see this problem.
Can you share the result of `pip list` command?

Anyway I think it's rather an Uvicorn issue, not FastAPI. So, it's better to look for answers there

**Reply by angad-singhh** [reply]
![image](https://github.com/tiangolo/fastapi/assets/108326021/bc10fc16-76c8-4a18-8cba-94dcce4ce4e5)
Please check

**Reply by Kludex** [reply]
It's probably a Windows issue. I think the issue already exists in Uvicorn.

**Reply by angad-singhh** [reply]
Okay, will check in that too!!

**Comment by angad-singhh** [comment]
I checked the uvicorn GitHub issues for this, found the fix resolve, the problem only exist in the current version of uvicorn==0.29.0 , rest it is working fine in any previous versions like 0.28.1 or 0.27.1 etc.
